navit/osd/core/osd_core.c

```diff
@@ -2291,10 +2292,10 @@ static struct osd_priv *osd_nav_toggle_announcer_new(struct navit *nav, struct o
         char **array;
         we = file_wordexp_new(attr->u.str);
         array = file_wordexp_get_array(we);
-        this->icon_src = g_strdup(array[0]);
+        this->icon_src = graphics_icon_path(array[0]);
         file_wordexp_destroy(we);
     } else
-        this->icon_src = graphics_icon_path("%s_32.xpm");
+        this->icon_src = graphics_icon_path("%s_48_48.png");
```

Comment:
Is there a better way than had coding .png with fixed size? Sailfish aims to be .svg only.

Reply:
This is just the fallback in case `icon_src` is not specified. The XSLT does specify an icon, though.